app/src/main/java/com/android/streetworkapp/ui/navigation/InfoDialogs.kt

```diff
@@ -0,0 +1,137 @@
+    private val topAppBarManager: TopAppBarManager?
+) {
+  private val screens = getScreens()
+
+  // Mapping Screen to InfoDialog
+  private val infoDialogs: MutableMap<String, InfoDialog> =
+      mutableMapOf(
+          "default" to defaultInfoDialog(),
+          screens.MAP to
+              InfoDialog(
+                  "Map",
+                  "How does the map work ?",
+                  "This is the map, here you can see each park that is near you. \n\n You can navigate to see more. Click on a park to see its details, such as their events !"),
+          screens.PROFILE to
+              InfoDialog(
+                  "Profile",
+                  "What is the Profile ?",
+                  "This is your profile page, you can see your friends and your score. \n\n You can also add friends using the 'Add friend' button and you can also access your settings by clicking the gear icon on the top right."),
+          screens.ADD_FRIEND to
+              InfoDialog(
+                  "AddFriend",
+                  "How do friends work ?",
+                  "This is where you can add a friend. \n\n To add, you need to have a friend nearby and activate bluetooth and location services. \n\n Once you are friends you will be able to do workouts with each other. \n\n Have fun together !"),
+          screens.PARK_OVERVIEW to
+              InfoDialog(
+                  "ParkOverview",
+                  "How does parks work ?",
+                  "This is the park overview, here you can see the park's details and the events that are planned in it. \n\n You can create an event or join an already existing one. You can also rate the park !"),
+          screens.ADD_EVENT to
+              InfoDialog(
+                  "AddEvent",
+                  "How to create an event ?",
+                  "This is your firsts steps towards creating an event ! \n\n You simply have to add a cool title, then, describe your event : What are you going to do there ? \n\n Finally, set the date and time and you are good to go !"),
+          screens.EVENT_OVERVIEW to
+              InfoDialog(
+                  "EventOverview",
+                  "What is the Event Overview ?",
+                  "This is the event overview, here you can access all of the information that you need. \n\n You are free to join the event if you want to participate !"),
+          screens.PROGRESSION to
+              InfoDialog(
+                  "Progression",
+                  "What is the Progression ?",
+                  "This is the progression screen, here you can see your progression and achievements. \n\n There is your record for each exercise in Training, click on it to see more details !")
+
+          // Auth not done because unnecessary
+          // Add more if needed
+          )
+
+  /** Returns the default InfoDialog (for screens that are not supported). */
+  fun defaultInfoDialog(): InfoDialog {
+    return InfoDialog(
+        "Default",
```

Comment:
Personally, I'd use the title **"Screen not implemented"** instead of **"Default"**, to make it more user-friendly. But that's just a suggestion, keep it that way if you prefer!

Reply:
Actually the first argument is the tag, which is only used for testTags purposes (or when no title is defined).

In this case the title is "Screen is not supported yet".

You can check this by commenting-out one element of the mapping and navigating there. That would make the call to the map return null, and the display will give the default parameters.

Reply:
You're right, thanks for the clarification!